Android/app/src/main/java/com/example/boardgamesocial/MainApp/Fragments/AddEventFragment.java

```diff
@@ -188,35 +304,104 @@ private void addEvent(Event event, String eventDateTime, String eventDescription
         });
     }
 
-    @Override
-    public void onItemClick(Bundle contextBundle) {
-        Game game = (Game) contextBundle.getSerializable(GameVH.GAME_KEY);
-        DataClsAdapter<Game, GameVH> dataClsAdapter = (DataClsAdapter<Game, GameVH>) recyclerView.getAdapter();
-        assert dataClsAdapter != null;
+    //Method to remove private games from the games collection
+    private Observable<JsonArray> fetchNonPrivateGames() {
 
-        RecyclerView.ViewHolder viewHolder = recyclerView
-                .findViewHolderForAdapterPosition(dataClsAdapter
-                        .getObjectList()
-                        .indexOf(game));
-        assert viewHolder != null;
+        Log.i(TAG, "Inside fetchNonPrivateGames");
 
-        CheckBox checkBox = viewHolder.itemView.findViewById(R.id.ck_hosted_game);
-        checkBox.setVisibility(View.INVISIBLE);
-        RelativeLayout relativeLayout = viewHolder.itemView.findViewById(R.id.hosted_game_card);
+        gameToUserList = new ArrayList<>();
 
-        if(Objects.nonNull(game)) {
-            Log.i(TAG, game.getId().toString());
-            if(checkBox.isChecked()) {
-                checkBox.setChecked(false);
-                gameIdList.remove(game.getId());
-                Log.i(TAG, "Removed - GameId: " + game.getId().toString());
-                relativeLayout.setBackgroundColor(Color.parseColor("#5EFF9B44"));
-            } else {
-                checkBox.setChecked(true);
-                gameIdList.add(game.getId());
-                Log.i(TAG, "Added - GameId: " + game.getId().toString());
-                relativeLayout.setBackgroundColor(Color.parseColor("#60000000"));
+        RetrofitClient.getClient().getCall(GameToUser.class, new HashMap<String, String>() {{
+            put("private", "True");
+        }}).blockingSubscribe(gameToUserJson -> gameToUserList = getObjectList(gameToUserJson, GameToUser.class));
```

Comment:
This will return all private games, did you mean to request private game on specific person?

Reply:
Currently, for a specific person, all games added by him are shown, even when it is public.